DNN Platform/Library/Application/DotNetNukeContext.cs

```diff
@@ -45,30 +44,27 @@ public static DotNetNukeContext Current
         {
             get
             {
-                if (_current == null)
+                if (current == null)
                 {
-                    _current = new DotNetNukeContext();
+                    current = Globals.DependencyProvider.GetRequiredService<IDnnContext>();
                 }
 
-                return _current;
+                return current is DotNetNukeContext context ? context : default(DotNetNukeContext);
             }
 
             set
             {
-                _current = value;
+                current = value;
             }
         }
 
         /// <summary>
         /// Gets get the application.
         /// </summary>
-        public Application Application
-        {
-            get
-            {
-                return this._application;
-            }
-        }
+        public Application Application { get => this.applicationInfo is Application app ? app : new Application(); }
```

Comment:
If the class is instantiated without Dependency Injection for some reason such as unit tests, I updated a default `Application` instantiation. Maybe we should update this to the `DependencyProvider`